File: ProjectEuler5/main.cpp
#include <iostream>
#include <chrono>
#include <ctime>

int main() {
    std::chrono::time_point<std::chrono::system_clock> start, end;
    start = std::chrono::system_clock::now();

    int numbers[] = {9699690, 20, 18, 16, 12};
    bool numberDivisible;
    int finalNumber = 0;

    do{
        finalNumber+=9699690;
        numberDivisible = true;
        for (int number : numbers) {
            if(finalNumber % number != 0) {
                numberDivisible = false;
                break;
            }
        }
    } while(!numberDivisible);

    end = std::chrono::system_clock::now();
    long elapsed_millis = std::chrono::duration_cast<std::chrono::milliseconds>(end-start).count();
    std::cout << "Elapsed Milliseconds: " << elapsed_millis << std::endl;

    std::cout << finalNumber << std::endl;
    return 0;
}

File: ProjectEuler5/src/main.rs
use std::time::{Duration, Instant};

fn gcd(mut a: i32, mut b: i32) -> i32 {
    let mut t;

    while b != 0 {
        t = b;
        b = a % b;
        a = t;
    }

    return a;
}

fn lcm(a: i32, b: i32) -> i32 {
    a*b/gcd(a,b)
}

fn main() {
    let now = Instant::now();

    let mut x = 2;
    let mut y;

    for i in 3..20 {
        y = i;
        x = lcm(x, y);
    }

    println!("Number: {}", x);
    println!("Elapsed µs: {}", now.elapsed().as_micros());
}
